### DISCUSSION TITLE
Crossover between 2 curves.

### DISCUSSION THREAD

**Discussion by RickPons** [mathnet/mathnet-numerics#962] [Q&A]
Hello, 

I am building an algo where I have 2 moving averages (curves) and I would like to get all the intersection points. I have an existing code to do this, but the performance is not good. 

I am very interested in this library, and it is looking very good and promising. I would like to know if someone could help me with some docs or how I could get this done using this amazing library! Thank you so much!

**Comment by jkalias** [comment]
I’m not sure that such functionality belongs in the library. I would though HIGHLY suggest the algorithm in this [paper](https://link.springer.com/content/pdf/10.1007/978-0-387-35490-3_21.pdf).

Its performance is on the order of O(n+m), where n and m is the total number of points for your two curves. It really has an impact when your curves have many points.
